README.md

```diff
@@ -1 +1,241 @@
+
+For example:
+
+Simply add the `-I` include path to your particular location
+of `avr-libstdcpp/include` on the command line...
+
+```sh
+avr-g++ -O2 -x c++ -I/my_path/avr-libstdcpp/include -mmcu=atmega328p test.cpp -o test.elf
+```
+
+... and seamlessly use standard library headers in your code.
+
+```C
+#include <array>
+#include <numeric>
+
+std::array<int, 3U> v {{ 1, 2, 3 }};
+
+int main()
+{
+  // 6
+  auto sum = std::accumulate(v.cbegin(), v.cend(), 0);
+
+  static_cast<void>(sum);
+}
+```
+
+Additional straightforward code samples exercising standard library usage
+can be found in the [examples](./examples) folder.
+
+## Guidance for tiny bare-metal systems
+
+### Very helpful go-to libraries
+
+In general the C++ standard library is intended
+to be written and implemented in a resource-sensitive
+fashion. This includes efforts to save on both
+memory as well as run-time. Algorithms are written
+with efficiency in mind and memory footprints are
+generally kept as minimal as possible.
+
+Some library components, however, are particularly well-suited
+for bare-metal microcontroller programming.
+These can be exceptionally helpful when used properly and sensibly
+in tiny bare-metal microcontroller environments.
+
+A subjective list of these the libraries/headers
+and their main uses includes, but is not limited to,:
+
+- `<array>` for containers having known, fixed size.
+- `<algorithm>` for standard algorithms such as sorting, minimax, sequential operations, etc.
+- `<cmath>` for projects requiring floating-point mathematical functions such as `std::sin()`, `std::exp()`, `std::frexp()` and many more.
+- `<cstdint>` which defines integral types having specified widths residing within `namespace` `std` like `std::uint8_t`.
+- `<limits>` offering compile-time query of numeric limits of built-in types.
+- `<numeric>` featuring a collection of useful numeric algorithms such ass `std::accumulate()`, etc.
+- `<type_traits>` for compile-time decisions based on types.
+
+With these libraries alone, the entire project can benefit
+from a great deal of the standard library's power without compromising
+in any way on performance or sleek memory footprint.
+This is because these libaries typically require no additional storage
+and their algorithms are top-notch regarding small footprint and performance.
+
+The following non-trivial, real-world example, for instance,
+wraps instances of an overly-simplified LED class abstraction
+as object-references in an `std::array`.
+Once stored, the application exercises the leds' toggle
+function in an algorithmic loop with `toggle()`-method call
+expressed via lambda function.
+
+```C
+#include <algorithm>
+#include <array>
+#include <type_traits>
+
+class led
+{
+public:
+  led() = default;
+
+  auto toggle() -> void { }
+};
+
+led led0;
+led led1;
+led led2;
+
+using led_ref_type = std::reference_wrapper<led>;
+
+std::array<led_ref_type, 3U> led_refs =
+{{
+  led0,
+  led1,
+  led2
+}};
```

Comment:
Same here

Reply:
Done. Will be in next commit.

Reply:
There's no need for an extra comment (we all get emails for each one), just resolve each  conversation.